internal/publish.go

```diff
@@ -170,16 +219,9 @@ func createTgz(composeContent []byte, appDir string) ([]byte, error) {
 
 		// Handle subdirectories
 		header.Name = strings.TrimPrefix(strings.Replace(file, appDir, "", -1), string(filepath.Separator))
-		if ignores != nil {
-			for _, ignore := range ignores {
-				if match, err := filepath.Match(ignore, header.Name); err == nil && match {
-					if !warned[ignore] {
-						fmt.Println("  |-> ignoring: ", ignore)
-					}
-					warned[ignore] = true
-					return nil
-				}
-			}
+
+		if skip, err := checkSkip(pm, fi, header.Name); skip {
```

Comment:
Is it ok if `skip` is false and `err` is not `nil`?

Reply:
i think you found an issue. I hate this function signature returning bool,error. But I want to keep the skip logic really close to docker so that its easier to feel confident we are doing the same thing as them.